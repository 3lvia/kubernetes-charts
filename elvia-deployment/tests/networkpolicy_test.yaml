suite: test networkpolicy
templates:
  - networkpolicy.yaml
tests:
  - it: network policy snapshot should match
    set:
      image.tag: latest
      name: demo-api
      namespace: core
      microserviceType: webapi
      environment: dev
      replicaCount: 2
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 400m
          memory: 500Mi
      accessPolicy:
        inbound: 
          rules:
            - application: "demo-app"
              namespace: "core"
        outbound:
          rules:
            - namespace: "core"
      ingress:
        subdomain: core
        path: "/demo-api"
      service:
        port: 8080

    asserts:
      - matchSnapshot: {}
