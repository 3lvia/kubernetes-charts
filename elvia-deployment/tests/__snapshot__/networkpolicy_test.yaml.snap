network policy snapshot should match:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      name: demo-api
      namespace: core
    spec:
      egress:
        - to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: core
        - to:
            - ipBlock:
                cidr: 0.0.0.0/0
                except:
                  - 10.0.0.0/8
                  - 192.168.0.0/16
                  - 172.16.0.0/12
        - to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: linkerd
        - to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: monitoring
              podSelector:
                matchLabels:
                  app.kubernetes.io/name: otel-collector
        - ports:
            - port: 53
              protocol: UDP
          to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: kube-system
              podSelector:
                matchLabels:
                  k8s-app: kube-dns
      ingress:
        - from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: core
              podSelector:
                matchLabels:
                  app.kubernetes.io/name: demo-app
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: dns
              podSelector:
                matchLabels:
                  app.kubernetes.io/name: traefik
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: monitoring
              podSelector:
                matchLabels:
                  app.kubernetes.io/name: prometheus
          ports:
            - port: 8080
              protocol: TCP
      podSelector:
        matchLabels:
          app: demo-api
      policyTypes:
        - Ingress
        - Egress
